internal/providers/yaml.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2024
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package providers
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"slices"
+
+	"gopkg.in/yaml.v3"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// GVK represents the GroupVersionKind structure in YAML.
+type GVK struct {
```

Comment:
nit: these types and functions (not methods ofc) might be unexported, I know this is the `internal` but still do not understand why these should be exposed if there are no callers outside the package

Reply:
personal preference